.github/workflows/inspektor-gadget.yml

```
@@ -697,6 +697,46 @@ jobs:
           rm -rf /tmp/.buildx-cache
           mv /tmp/.buildx-cache-new /tmp/.buildx-cache
 
+  analyze-and-build:
+    name: IG binary size analysis
+    runs-on: ubuntu-latest
+    needs: 
+      - build-ig 
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@3041bf56c941b39c61721a86cd11f3bb1338122a # v5.2.0
+        with:
+          go-version: '1.23.4'
+          cache: true
+        id: go
+      - name: Install go-size-analyzer
+        run: |
+          go install github.com/Zxilly/go-size-analyzer/cmd/gsa@e47aa2561e70b937f2be7441b2eeca984ff60e8b
+      - name: Get ig-linux-amd64.tar.gz from artifact.
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: ig-linux-amd64-tar-gz
+          path: /home/runner/work/inspektor-gadget/
+      - name: Unpack ig-linux-amd64.tar.gz
+        run: |
+          tar zxvf /home/runner/work/inspektor-gadget/ig-linux-amd64.tar.gz
+      - name: Generate JSON
+        run: |
+          gsa ./ig -f "json" --compact > out.json
+      - name: Extract Packages as JSON Array
+        run: |
+          jq '[.packages | to_entries[] | {package: .key, size: .value.size}] | sort_by(.size) | reverse' out.json > packages.json
+        shell: bash
+      - name: Generate Markdown Table
+        uses: buildingcash/json-to-markdown-table-action@v1.1.0
+        id: table
+        with:
+          json_file_path: packages.json
+      - name: Add Markdown Table
+        run: |
+          echo '### Package Size Summary' >> $GITHUB_STEP_SUMMARY
+          echo "${{ steps.table.outputs.table }}" >> $GITHUB_STEP_SUMMARY
```

The expressions inside of `${{ }}` are evaluated and substituted with the resulting values before the shell script is run, so if the output mistakenly contains a payload like `a"; command`, then it would do something unexpected.

Github recommends the following code pattern to avoid this issue:

```suggestion
      - name: Add Markdown Table
        env:
          TABLE_OUTPUT=${{ steps.table.outputs.table }}
        run: |
          echo '### Package Size Summary' >> $GITHUB_STEP_SUMMARY
          echo "$TABLE_OUTPUT" >> $GITHUB_STEP_SUMMARY
```

cc @mauriciovasquezbernal

---

Getting this error:
> 737 |           TABLE_OUTPUT=${{ steps.table.outputs.table }}
    |           ^~~~~~~~~~~~~~~~
Error: Process completed with exit code 1.

I think it should have been like:
```
env:
  TABLE_OUTPUT: ${{ steps.table.outputs.
```
what do you think @alban ?